scrapy/mail.py

```diff
@@ -36,28 +38,18 @@ def _to_bytes_or_none(text):
     return to_bytes(text)
 
 
-class MailSender(object):
-
-    def __init__(self, smtphost='localhost', mailfrom='scrapy@localhost',
-            smtpuser=None, smtppass=None, smtpport=25, smtptls=False, smtpssl=False, debug=False):
-        self.smtphost = smtphost
-        self.smtpport = smtpport
-        self.smtpuser = _to_bytes_or_none(smtpuser)
-        self.smtppass = _to_bytes_or_none(smtppass)
-        self.smtptls = smtptls
-        self.smtpssl = smtpssl
-        self.mailfrom = mailfrom
-        self.debug = debug
+class BaseMailSender(object):
 
     @classmethod
     def from_settings(cls, settings):
-        return cls(settings['MAIL_HOST'], settings['MAIL_FROM'], settings['MAIL_USER'],
-            settings['MAIL_PASS'], settings.getint('MAIL_PORT'),
-            settings.getbool('MAIL_TLS'), settings.getbool('MAIL_SSL'))
+        raise NotImplementedError
 
     def send(self, to, subject, body, cc=None, attachs=(), mimetype='text/plain', charset=None, _callback=None):
+        raise NotImplementedError
+
+    def _get_message(self, to, subject, body, cc=None, attachs=(), mimetype='text/plain', charset=None):
         if attachs:
-            msg = MIMEMultipart()
+            msg = MIMEMultipart('alternative')
```

Comment:
Why is this needed? Is it related to SESMailSender?

Reply:
Not needed. Only if we want to have an alternative text version of the e-mail content together with the HTML version. I don't think it is needed here.